src/data/simulator.py

```
@@ -0,0 +1,388 @@
+
+            if should_trigger_push(last_visit_day, day, last_push_day, marketing_cfg):
+                last_push_day = day
+                response = persona_cfg["marketing_response"]
+                roll = rng.random()
+                if roll < response["push_notification"]:
+                    active_churn_suppressed_until = max(active_churn_suppressed_until, day + 7)
+
+        # --- Visit-driven events ---
+        if day in visit_days:
+            last_visit_day = day
+            # Number of events on this visit (1–5)
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)